sig/datadog/core/rate_limiter.rbs

```diff
@@ -1,17 +1,42 @@
 module Datadog
   module Core
     class RateLimiter
-      def allow?: (untyped size) -> nil
+      def allow?: (?::Integer size) -> (false | true)
+
       def effective_rate: () -> nil
```

Comment:
Minor: This is an interesting one... `nil` is more of a "what it is right now" because this is a bit of an abstract method. Should we perhaps use `-> void` to mark that this shouldn't be used by itself? Or `Float?`